### PR TITLE
Corrected colouring on percentile and permutation plots for Terms, rdiff, RR and OR

### PR DESCRIPTION
**Pull request by mark-edmondson-jones**

### CODE REVIEW

File: R/plotpercentile.R [hunk 1]
```diff
@@ -40,31 +42,46 @@ plotpercentile <- function(x){
...
+    p<- ggplot2::ggplot(data=plotdata[plotdata$type=="Permutation", ], aes(x=x, y=y, group=label), aspect="iso")
+    if (coloring %in% c("p","p.adj","fish")) {
+      p <- p +
+        ggplot2::scale_colour_gradientn("log(p-val)",
+                                        colours=palette$grpalette,
+                                        values = palette$values,
+                                        limits = palette$limits
+       )
+    } else if (coloring != "Terms") {
+      p <- p +
```
Comment (stefanoborini) [inline R/plotpercentile.R]: Whatever editor you are using, it's leaving trailing spaces at the end of lines. If you are using RStudio, there's an option "String trailing horizontal whitespace when saving" to enable under "Options - Code - Saving".

This is to reduce the chance of conflict due to changes in spaces (most editor nowadays have the option on by default)

File: R/plotpercentile.R [hunk 2]
```diff
@@ -40,31 +42,46 @@ plotpercentile <- function(x){
...
+        ggplot2::scale_colour_gradient(coloring)
+    }
+    p <- p +
+      ggplot2::geom_path(color="grey80") +
+      ggplot2::geom_path(data=plotdata[plotdata$type=="Percentile", ], aes(x=x, y=y, group=label), color="grey40", size=1.25)
+    if (coloring != "Terms") {
+      p <- p +
```
Comment (stefanoborini) [inline R/plotpercentile.R]: This condition and the else differ only in the color that you are passing. There's no point in duplicating the complete set of lines. Just set the color and then pass it as a variable.

File: R/plotpermutation.R [hunk 3]
```diff
@@ -30,27 +32,40 @@ plotpermutation <- function(x){
...
-      ggplot2::coord_fixed(ratio=1) +
-      ggplot2::ggtitle(Title) +
-      ggplot2::annotate(geom="text", x=data2.labels$xpos, y=data2.labels$ypos,
+      )
+  } else if (coloring != "Terms") {
+    p <- p + ggplot2::scale_colour_gradient(coloring)
+  }
+  p <- p + ggplot2::geom_path(color = "Grey80")
+  if (coloring!="Terms") {
+    p <- p +
```
Comment (stefanoborini) [inline R/plotpermutation.R]: same as before.

File: R/plotpercentile.R (hunk 2 above)
Comment (mark-edmondson-jones) [reply R/plotpercentile.R]: Not sure which lines you are referring to.  Lines 62-63 set colour as an aesthetic using a variable (color).  Lines 66-67 specify a fixed color ("blue").

File: R/plotpermutation.R (hunk 3 above)
Comment (mark-edmondson-jones) [reply R/plotpermutation.R]: As before: in one case color appears within the aes brackets and in the other it does not.

File: R/plotpercentile.R (hunk 2 above)
Comment (stefanoborini) [reply R/plotpercentile.R]: Can't you set the fixed "blue" color in the variable if unspecified, and invoke the plot lines only once, always with the variable?

Comment (mark-edmondson-jones) [reply R/plotpercentile.R]: I don't know how ggplot would interpret color being defined both inside and outside aes(),  I think that this approach is safer and more transparent.

**Pull request merged**